clients/imodels-client-management/src/base/internal/AxiosRestClient.ts

```diff
@@ -20,59 +20,78 @@ export class AxiosRestClient implements RestClient {
     const requestConfig: AxiosRequestConfig = {
       headers: params.headers
     };
 
     if (params.responseType === ContentType.Png) {
       requestConfig.responseType = "arraybuffer";
-      const responseData: Buffer | ArrayBuffer = await this.executeRequest(async () => axios.get(params.url, requestConfig));
-      if (responseData instanceof ArrayBuffer)
-        return new Uint8Array(responseData);
-      return responseData;
+      const response = await this.executeRequest(async () => axios.get(params.url, requestConfig));
+
+      const data: Buffer | ArrayBuffer = response.data;
+      if (data instanceof ArrayBuffer)
+        return { ...response, data: new Uint8Array(data) };
+
+      return response;
     }
 
     return this.executeRequest(async () => axios.get(params.url, requestConfig));
   }
 
-  public async sendPostRequest<TResponse>(params: HttpRequestWithJsonBodyParams): Promise<TResponse> {
+  public async sendPostRequest<TData>(params: HttpRequestWithJsonBodyParams): Promise<HttpResponse<TData>> {
     const requestConfig: AxiosRequestConfig = {
       headers: params.headers
     };
 
     return this.executeRequest(async () => axios.post(params.url, params.body.content ?? {}, requestConfig));
   }
 
-  public async sendPutRequest<TResponse>(params: HttpRequestWithBinaryBodyParams): Promise<TResponse> {
+  public async sendPutRequest<TData>(params: HttpRequestWithBinaryBodyParams): Promise<HttpResponse<TData>> {
     const requestConfig: AxiosRequestConfig = {
       headers: params.headers
     };
 
     return this.executeRequest(async () => axios.put(params.url, params.body.content, requestConfig));
   }
 
-  public async sendPatchRequest<TResponse>(params: HttpRequestWithJsonBodyParams): Promise<TResponse> {
+  public async sendPatchRequest<TData>(params: HttpRequestWithJsonBodyParams): Promise<HttpResponse<TData>> {
     const requestConfig: AxiosRequestConfig = {
       headers: params.headers
     };
 
     return this.executeRequest(async () => axios.patch(params.url, params.body.content ?? {}, requestConfig));
   }
 
-  public async sendDeleteRequest<TResponse>(params: HttpRequestParams): Promise<TResponse> {
+  public async sendDeleteRequest<TData>(params: HttpRequestParams): Promise<HttpResponse<TData>> {
     const requestConfig: AxiosRequestConfig = {
       headers: params.headers
     };
+
     return this.executeRequest(async () => axios.delete(params.url, requestConfig));
   }
 
-  private async executeRequest<TResponse>(requestFunc: () => Promise<AxiosResponse<TResponse>>): Promise<TResponse> {
+  private async executeRequest<TData>(requestFunc: () => Promise<AxiosResponse<TData>>): Promise<HttpResponse<TData>> {
     try {
       const response = await requestFunc();
-      return response.data;
+
+      return {
+        data: response.data,
+        headers: {
+          get: (headerName: string) => {
```

Comment:
wouldn't it be easier to have an adapter class, like `AxiosHeadersAdapter implements HttpResponseHeaders ...`, which would accept the axios headers as a constructor argument? I do not see why we need to dynamically construct functions here.

Reply:
Created and passed through constructor a factory for `HttpResponseHeaders`. Was a bit on the line whether to pass it through constructor or just create it directly, since it does not seem like a dependency, more like internals. What do you think?

Reply:
I do not have a strong opinion of this, we can leave it with a factory, looks a bit cleaner a bit.